### PR TITLE
Fix Code Analysis warnings

### PR DESCRIPTION
**Pull request by conniey**
This fixes warnings that an upgrade to Microsoft.CodeAnalysis.FxCopAnalyzers 2.6.0 will discover after we migrate to it.

### CODE REVIEW

File: src/ApiPort/ConsoleProgressReporter.cs [hunk 1]
```diff
@@ -89,17 +89,28 @@ private static void WriteColorLine(string message, ConsoleColor color)
 
         public void Dispose()
         {
-            if (!_disposed)
+            Dispose(true);
```
Comment (twsouthwick) [inline src/ApiPort/ConsoleProgressReporter.cs]: Another option here would be to seal the type. Is that possible?

File: src/Microsoft.Fx.Portability/TextWriterProgressReporter.cs [hunk 2]
```diff
@@ -41,6 +42,12 @@ public void Resume()
         { }
 
         public void Dispose()
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability/TextWriterProgressReporter.cs]: As mentioned above, maybe sealing the type would work

File: tests/ApiPort.Tests/AnalyzeOptionsTests.cs [hunk 3]
```diff
@@ -31,15 +31,15 @@ public void TestAssemblyFlag_Directory()
...
+        public static void NoArgs()
         {
             var options = CommandLineOptions.ParseCommandLineOptions(Array.Empty<string>());
 
             Assert.Equal(AppCommand.Exit, options.Command);
         }
 
         [Fact]
-        public void AnalyzeNoFile()
+        public static void AnalyzeNoFile()
```
[the remark this answers is not in the document]
Comment (conniey) [reply tests/ApiPort.Tests/AnalyzeOptionsTests.cs]: Yes. I thought it was magical too that the tests were still found and running.

File: src/ApiPort/ConsoleProgressReporter.cs (hunk 1 above)
Comment (conniey) [reply src/ApiPort/ConsoleProgressReporter.cs]: Yes. Sealing the type would be a lot easier. I'll do that for both `ProgressReporters`

**Comment by conniey** [comment]
@twsouthwick  I verified that the VS Extension still works in VS2017 with the changes. Is everything OK?

**Pull request merged**